### PR TITLE
ACMS-3461: Test case for settings class.

### PR DESCRIPTION
**Pull request by apathak18**
**Motivation**

Fixes ACMS-3461

**Proposed changes**
Test coverage for settings and plugin

### CODE REVIEW

File: tests/fixtures/project/docroot/sites/default/settings/test.settings.php [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?php
+
+/**
+ * @file
+ * Only for testing purpose.
+ */
```
Comment (deepakmishra2) [inline tests/fixtures/project/docroot/sites/default/settings/test.settings.php:6]: what is the purpose of having this empty file?

File: tests/fixtures/project/docroot/sites/settings/testing.php [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<?php
+
+/**
+ * @file
+ * Only for testing purpose.
+ */
```
Comment (deepakmishra2) [inline tests/fixtures/project/docroot/sites/settings/testing.php:6]: Similarly this

File: tests/src/unit/PluginUnitTest.php [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  /**
+   * The recommended settings object.
+   */
+  protected Plugin $plugin;
+
+  /**
+   * The Composer service.
+   */
+  protected Composer $composer;
+  /**
```
Comment (deepakmishra2) [inline tests/src/unit/PluginUnitTest.php:22]: add extra line here

File: tests/fixtures/project/docroot/sites/settings/testing.php (hunk 2 above)
Comment (apathak18) [reply tests/fixtures/project/docroot/sites/settings/testing.php:6]: Fixture directory isn't added in the PR so one temp file is required.

**Review by vishalkhode1 — APPROVED** [review]

**Review by rajeshreeputra — APPROVED** [review]

**Pull request merged**